localstack/services/events/target.py

```diff
@@ -0,0 +1,304 @@
+
+    @abstractmethod
+    def send_event(self):
+        pass
+
+    def _validate_input(self, target: Target):
+        """Provide a default implementation that does nothing if no specific validation is needed."""
+        pass
+
+    def _initialize_clients(self) -> ServiceLevelClientFactory:
+        """Initializes AWS service clients, with or without assuming a role of service source.
+        If a role from a target is provided, the client will be initialized with the assumed role and events service principal.
+        If no role is provided e.g. calling send_events directly, the client will be initialized with the account ID and region."""
+        if role_arn := self.target.get("role_arn"):
+            clients = connect_with_role(role_arn, self.region)
+            if not clients:
+                clients = connect_to(aws_access_key_id=self.account_id, region_name=self.region)
+        else:
+            clients = connect_to(aws_access_key_id=self.account_id, region_name=self.region)
+        return clients
+
+
+TargetWorkerDict = dict[Arn, TargetWorker]
+
+
+class ApiGatewayTargetWorker(TargetWorker):
+    def send_event(self, event):
+        raise NotImplementedError("ApiGateway target is not yet implemented")
+
+
+class AppSyncTargetWorker(TargetWorker):
+    def send_event(self, event):
+        raise NotImplementedError("AppSync target is not yet implemented")
+
+
+class BatchTargetWorker(TargetWorker):
+    def send_event(self, event):
+        raise NotImplementedError("Batch target is not yet implemented")
+
+    def _validate_input(self, target: Target):
+        if not collections.get_safe(target, "$.BatchParameters.JobDefinition"):
+            raise ValueError("BatchParameters.JobDefinition is required for Batch target")
+        if not collections.get_safe(target, "$.BatchParameters.JobName"):
+            raise ValueError("BatchParameters.JobName is required for Batch target")
+
+
+class ContainerTargetWorker(TargetWorker):
+    def send_event(self, event):
+        raise NotImplementedError("ECS target is not yet implemented")
+
+    def _validate_input(self, target: Target):
+        super()._validate_input(target)
+        if not collections.get_safe(target, "$.EcsParameters.TaskDefinitionArn"):
+            raise ValueError("EcsParameters.TaskDefinitionArn is required for ECS target")
+
+
+class EventsTargetWorker(TargetWorker):
+    def send_event(self, event):
+        events_client = self.clients.events.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
+        )
+        eventbus_name = self.target["Arn"].split(":")[-1].split("/")[-1]
+        detail = event.get("detail") or event
+        resources = event.get("resources") or [self.rule_arn] if self.rule_arn else []
+        events_client.put_events(
+            Entries=[
+                {
+                    "EventBusName": eventbus_name,
+                    "Source": event.get("source", self.service) or "",
```

Comment:
copy pasted that from our v1 provider, aligned it in, further testing is needed, because from the docs (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/events/client/put_events.html) I would argue there should be a validation that fails if not all three fields are provided Detail, DetailType, and Source
preliminary fix in: https://github.com/localstack/localstack/pull/10613/commits/c7a419352779797ae8259200858463cf0a04b47b